cmd/dwi2adc.cpp

```diff
@@ -88,25 +162,21 @@ void run () {
     ++dwi_axis;
   INFO ("assuming DW images are stored along axis " + str (dwi_axis));
 
-  Eigen::MatrixXd b (grad.rows(), 2);
-  for (ssize_t i = 0; i < b.rows(); ++i) {
-    b(i,0) = 1.0;
-    b(i,1) = -grad (i,3);
-  }
-
-  auto binv = Math::pinv (b);
+  auto opt = get_options ("ivim");
+  bool ivim = opt.size();
+  int bmin = get_option_value("cutoff", 120);
 
   Header header (dwi);
   header.datatype() = DataType::Float32;
   DWI::stash_DW_scheme (header, grad);
   PhaseEncoding::clear_scheme (header);
   header.ndim() = 4;
-  header.size(3) = 2;
+  header.size(3) = ivim ? 4 : 2;
```

Comment:
So for non-IVIM, the output is still a 4D image with two volumes, being the predicted S0 and ADC? Personally I'd prefer the output in this scenario to be a 3D image containing ADC, and requesting the predicted S0 output should be a command-line option and a separate image. It's weird opening up the image that is the output of a command called "`dwi2adc`" and seeing what looks like a DWI.

Reply:
#2804 is listing for dimensionality of output data for `dwi2adc`; did not realise it was not within scope of #2195. But this PR might nevertheless be the right context to contemplate.

Reply:
This PR (#2578) does not change the interface for dwi2adc: the output has always been a 4D image with 2 volumes as far as I can remember. I don't particularly like this either and we can think about changing the interface later, but the focus of this PR is on IVIM fitting.

Reply:
Omitting interface change for the sake of getting the PR though; already struggling with getting PRs resolved.